src/main/java/org/opensearch/geospatial/index/mapper/xypoint/XYPoint.java

```diff
@@ -61,4 +81,156 @@ public void resetCoords(double x, double y) {
     public Point asGeometry() {
         return new Point(getX(), getY());
     }
+
+    /**
+     * Set x and y coordinates of XYPoint if input contains WKT or coordinates.
+     *
+     * @param value  input String which needs to be parsed and validated
+     * @param ignoreZValue  boolean parameter which decides if z coordinate needs to be ignored or not
+     * @return XYPoint after setting the x and y coordinates
+     */
+    public XYPoint resetFromString(String value, final boolean ignoreZValue) {
+        if (value.toLowerCase(Locale.ROOT).contains(POINT_PRIMITIVE)) {
+            return resetFromWKT(value, ignoreZValue);
+        }
+        return resetFromCoordinates(value, ignoreZValue);
+    }
+
+    /**
+     * Set x and y coordinates of XYPoint if input contains coordinates.
+     *
+     * @param value  input String which contains coordinates that needs to be parsed and validated
+     * @param ignoreZValue  boolean parameter which decides if z coordinate needs to be ignored or not
+     * @return XYPoint after setting the x and y coordinates
+     * throws OpenSearchParseException
+     */
+    public XYPoint resetFromCoordinates(String value, final boolean ignoreZValue) {
```

Comment:
shall we check for null in all public methods?

Reply:
updated